apps/src/lib/node/ledger/events/log/dumb_queries.rs

```diff
@@ -6,46 +6,59 @@
 //! tm.event='NewBlock' AND <accepted|applied>.<$attr>='<$value>'
 //! ```
 
-use namada::types::hash::HexEncodedHash;
+use std::marker::PhantomData;
+use namada::types::hash::Hash;
 
 use crate::node::ledger::events::{Event, EventType};
 
 /// A [`QueryMatcher`] verifies if a Namada event matches a
 /// given Tendermint query.
 #[derive(Debug, Clone)]
-pub struct QueryMatcher<'q> {
+pub struct QueryMatcher<'a, T>
+where
+    T: 'a +  PartialEq + TryFrom<&'a str>
+{
```

Comment:
I think it's cleaner to leave trait bounds to impl blocks

Reply:
i.e. remove the trait bounds here

Reply:
Fair enough. Can literally just remove that and the code will work.